yarn-project/noir-contracts/src/contracts/non_native_token_contract/src/main.nr

```diff
@@ -113,7 +113,7 @@ contract NonNativeToken {
         inputs: PrivateContextInputs,
         //*********************************/
         amount: Field,
-        sender: Field,
+        sender: Field, // TODO: Should verify sender.
```

Comment:
What does this 'TODO' mean, sorry? Verify how?

Reply:
Cause this allows any account to be the "sender": if I know your decrypted notes I can spend them. I've changed everywhere else to use msg_sender as the note spender. But this is called by a another contract (swap). So will have to think about how to change the flow (or use `tx_origin` 😂).